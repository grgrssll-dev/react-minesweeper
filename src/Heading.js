import React, { PureComponent } from 'react';
import './Heading.css';

import GameValue from './GameValue';

class Heading extends PureComponent {

	render() {
		console.log('-- heading render');
		const { minesRemaining, timeElapsed, isGameOver, onGameStart } = this.props;

		return (
			<div className="heading">
				<div className="heading--mines-remaining">
					<GameValue value={minesRemaining} />
				</div>
				<div className="heading--spread">
					<span className="heading--game-indicator"
						role="img"
						aria-label={isGameOver ? 'New Game' : 'Game in progress'}
						onClick={onGameStart}>{isGameOver ? 'ðŸ˜Ž' : 'ðŸ˜€'}</span>
				</div>
				<div className="heading--time-elapsed">
					<GameValue value={timeElapsed} />
				</div>
			</div >
		);
	}
}

export default Heading;
