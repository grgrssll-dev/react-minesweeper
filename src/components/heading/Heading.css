.heading {
	border-bottom-color: var(--game-border-light);
	border-left-color: var(--game-border-dark);
	border-right-color: var(--game-border-light);
	border-style: solid;
	border-top-color: var(--game-border-dark);
	border-width: 0.125rem;
	padding: 0.25rem;
}

.heading--mines-remaining {
	display: inline-block;
	vertical-align: middle;
}

.heading--time-elapsed {
	display: inline-block;
	vertical-align: middle;
}

.heading--spread {
	display: inline-block;
	vertical-align: middle;
	width: calc(100% - 5rem);
}

.heading--game-indicator {
	cursor: default;
	display: block;
	border-width: 0.125rem;
	border-style: solid;
	border-top-color: var(--game-border-light);
	border-left-color: var(--game-border-light);
	border-right-color: var(--game-border-dark);
	border-bottom-color: var(--game-border-dark);
	box-sizing: border-box;
	font-size: 1rem;
	height: 1.5rem;
	line-height: 1;
	margin: 0 auto;
	padding: 0.125rem;
	text-align: center;
	width: 1.5rem;
}