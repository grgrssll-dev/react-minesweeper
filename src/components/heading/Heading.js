import React from 'react';
import './Heading.css';

import GameValue from '../gameValue/GameValue';

function heading(props) {
	console.log('-- heading render');
	const { minesRemaining, timeElapsed, isGameOver, onGameStart } = props;

	return (
		<div className="heading">
			<div className="heading--mines-remaining">
				<GameValue value={minesRemaining} />
			</div>
			<div className="heading--spread">
				<span className="heading--game-indicator"
					role="img"
					aria-label={isGameOver ? 'New Game' : 'Game in progress'}
					onClick={onGameStart}>{isGameOver ? 'ðŸ˜Ž' : 'ðŸ˜€'}</span>
			</div>
			<div className="heading--time-elapsed">
				<GameValue value={timeElapsed} />
			</div>
		</div >
	);
}

export default heading;
