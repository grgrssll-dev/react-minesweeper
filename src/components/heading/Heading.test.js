import React from 'react';
import { shallow } from 'enzyme/build';
import Heading from './Heading';

const baseProps = {
	minesRemaining: 10,
	timeElapsed: 4,
	isGameOver: jest.fn(),
	onGameStart: jest.fn(),
	className: 'class-name',
};

describe('Heading [Unit]', () => {
	beforeEach(() => {
		jest.clearAllMocks();
	});

	it('should match component snapshot', () => {
		expect.assertions(1);
		const props = {
			...baseProps,
		};
		const wrapper = shallow(<Heading {...props} />);
		expect(wrapper).toMatchSnapshot();
	});
});