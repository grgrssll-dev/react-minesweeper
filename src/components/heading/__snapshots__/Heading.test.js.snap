// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading [Unit] should match component snapshot 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
